### PR TITLE
GHA release pipeline

### PR DESCRIPTION
**Pull request by maxrichie5**
### Background

We currently have to manually release PAT from our local machines. This creates a github workflow to release it.

Please note that the pypi password has NOT been added to the repo yet, waiting until team approves of this

### Changes

* Adds release workflow to release changes when merged to master branch
* Cleans up ci workflow
* Cleans up makefile
* Updates version logic to match what `mypy` does
* Updates version of black (LOTS of changes because I ran `make fmt`)

### Testing

* Ran the test dist steps locally

### CODE REVIEW

**Review by cdzombak — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -1,8 +1,5 @@
-from pathlib import Path
 from setuptools import setup, find_packages
-
-this_directory = Path(__file__).parent
-version = (this_directory / "VERSION").read_text().strip()
```
Comment (darwayne) [inline setup.py:5]: was this no bueno?

**Review by darwayne — COMMENTED** [review]
Pretty cool stuff, curious as to what the new workflow will be with this change ... every PR change we need to bump the version.py file? .. if we forget will CI fail? (that'd be awesome)

File: setup.py (hunk 1 above)
Comment (darwayne) [reply setup.py:5]: I really liked the VERSION file concept more so than the version.py concept

Comment (maxrichie5) [reply setup.py:5]: This didn't allow us to use the version number within the module. So we still had to have the version number in two places. I looked at how `mypy` does version numbers and copied them for the most part

Comment (darwayne) [reply setup.py:5]: gotcha ok

**Review by darwayne — APPROVED** [review]

**Review by darwayne — APPROVED** [review]

**Comment by grantjoy** [comment]
@maxrichie5 is there a reason this didn't merge after @darwayne approved it?

**Comment by maxrichie5** [comment]
> @maxrichie5 is there a reason this didn't merge after @darwayne approved it?

Umm. Maybe the secret mentioned in the desc? I honestly don't remember

**Pull request closed without merging**